src/Resource.js

```diff
@@ -87,28 +100,34 @@ module.exports = class Resource {
     this.vocabContentTypeHeaderFallback = vocabContentTypeHeaderFallback;
   }
 
-  async processInputs(config, processInputsCallback) {
-    debug(`Processing input resources: [${this.inputResources}]...`);
-    const datasetsPromises = this.inputResources.map((inputResource) => {
-      return Resource.readResource(
-        inputResource,
-        this.vocabAcceptHeaderOverride,
-        this.vocabContentTypeHeaderOverride,
-        this.vocabContentTypeHeaderFallback
-      ).catch((rootCause) => {
-        return Resource.attemptToReadGeneratedResource(
-          config,
+  async processInputs(config) {
+    const datasets = [];
+
+    for (let i = 0; i < this.inputResources.length; i++) {
+      const inputResource = this.inputResources[i];
+      try {
+        const resource = await Resource.readResourceViaCache(
           inputResource,
-          rootCause
+          this.vocabAcceptHeaderOverride,
+          this.vocabContentTypeHeaderOverride,
+          this.vocabContentTypeHeaderFallback
         );
-      });
-    });
 
-    const datasets = await Promise.all(datasetsPromises);
+        datasets.push(resource);
```

Comment:
So I don't think so (not with the current code structure anyways). The reason isn't so much the 'number' of concurrent reads, it's that we have a loop for multiple 'forms' of an artifact (e.g., a Java string-based one, a JS VocabTerm-based one, a TS string-based one, a JS RDF-library-based one, etc.). So if we attempt to parallelize that at all, even just with a threadpool of 2, both of those would be reading the same source vocab multiple times.
So by serializing instead, we just read the source vocab once, store it in our cache, and then the next 'form' of that artifact from the same source vocab reads from the cache instead.
To intorduce parallelism again (which I don't think is particulary important for the AG, but I guess would be 'nice'), we'd have to restructure the nested looping structure we have now I think, but to basically still serialize the artifact-form generation ('cos we should still only be readng the source vocab just once).
And then, even for completely different artifacts, we could potentially be reading from the same source vocab, in which case parallelising would again introduce the potential of re-reading a source vocab multiple times.
So perhaps trying to parallelise at all might require a complete pass over the entire set of YAMLs to determine the entire set of source vocabs, which we'd load into a cache, and then once all are loaded, then run the loops of artifact generations. But again, probably a pretty major refactoring of the current code structure I think, but interesting...
And since the linter isn't complaining at the moment (maybe it's not configured correctly!?), I'm happy to leave this as it is for now...